src/client/Client.cc

```diff
@@ -8781,19 +8781,19 @@ int Client::_write(Fh *f, int64_t offset, uint64_t size, const char *buf,
 
   // copy into fresh buffer (since our write may be resub, async)
   bufferlist bl;
-  bufferptr *bparr = NULL;
+  std::vector<bufferptr> bpv;
   if (buf) {
       bufferptr bp;
       if (size > 0) bp = buffer::copy(buf, size);
       bl.push_back( bp );
   } else if (iov){
       //iov case 
-      bparr = new bufferptr[iovcnt];
+      bpv.resize(iovcnt);
```

Comment:
I think you can but it's a C++14 thing according to:

http://en.cppreference.com/w/cpp/container/vector/vector